backend/src/main/scala/sbt/internal/inc/bloop/ZincInternals.scala

```diff
@@ -65,16 +65,12 @@ object ZincInternals {
   import sbt.internal.inc.JavaInterfaceUtil.EnrichOptional
   object ZincExistsStartPos {
     def unapply(position: Position): Option[(Int, Int)] = {
-      // Use the range position if available first. Otherwise fallback in pointer information.
-      val rangePosition = position.startLine.toOption.flatMap(
-        startLine =>
-          position.startColumn().toOption.map(startColumn => (startLine.toInt, startColumn.toInt))
-      )
-
-      rangePosition.orElse {
-        position.line.toOption
-          .flatMap(line => position.pointer.toOption.map(column => (line.toInt, column.toInt)))
-      }
+      // Javac doesn't provide column information, so we just assume beginning of the line
+      val rangePosition = for { startLine <- position.startLine().toOption.map(_.toInt) }
```

Comment:
Can we please use the previous style with map/flatmaps instead to minimise the change diff as much as possible?